File: data/model/src/main/java/com/harera/model/model/Post.java
package com.harera.model.model;

import com.google.firebase.Timestamp;
import com.google.firebase.firestore.IgnoreExtraProperties;

@IgnoreExtraProperties
public final class Post {
    public String postImageUrl;
    public Timestamp time;
    public String caption;
    public String postId;
    public String uid;
    public String profileName;
    public String profileImageUrl;
    public int likesNumber;
    public int commentsNumber;

    public Post(String postImageUrl, Timestamp time, String caption, String postId, String uid, String profileName, String profileImageUrl, int likesNumber, int commentsNumber) {
        this.postImageUrl = postImageUrl;
        this.time = time;
        this.caption = caption;
        this.postId = postId;
        this.uid = uid;
        this.profileName = profileName;
        this.profileImageUrl = profileImageUrl;
        this.likesNumber = likesNumber;
        this.commentsNumber = commentsNumber;
    }

    public Post() {
    }
}

File: data/model/src/main/java/com/harera/model/model/Post.kt
package com.harera.model.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
@Entity
data class Post(
    @PrimaryKey
    val postId: Int,
    val caption: String,
    val username: String,
    val type: Int,
    var postImageUrl: String,
    val time: String,
)
